File: frontend/src/types/skill.ts
export interface Skill {
  id: string
  name: string
  display_name: string | null
  description: string | null
  skill_type: 'llm' | 'rag' | 'mcp'
  config: string
  trigger_keywords: string
  is_active: boolean
  priority: number
  created_at: string
  updated_at: string
}

export interface SkillCreate {
  name: string
  display_name?: string
  description?: string
  skill_type?: string
  config?: string
  trigger_keywords?: string
  is_active?: boolean
  priority?: number
}

export interface SkillUpdate {
  display_name?: string
  description?: string
  skill_type?: string
  config?: string
  trigger_keywords?: string
  is_active?: boolean
  priority?: number
}

File: backend/app/schemas/skill.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class SkillCreate(BaseModel):
    name: str
    display_name: Optional[str] = None
    description: Optional[str] = None
    skill_type: str = "llm"
    config: str = "{}"
    trigger_keywords: str = "[]"
    is_active: bool = True
    priority: int = 100


class SkillUpdate(BaseModel):
    display_name: Optional[str] = None
    description: Optional[str] = None
    skill_type: Optional[str] = None
    config: Optional[str] = None
    trigger_keywords: Optional[str] = None
    is_active: Optional[bool] = None
    priority: Optional[int] = None


class SkillRead(BaseModel):
    id: str
    name: str
    display_name: Optional[str] = None
    description: Optional[str] = None
    skill_type: str
    config: str
    trigger_keywords: str
    is_active: bool
    priority: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
